_posts/2021-01-12-updating-foreman-inventory-with-system-facts.md

```diff
@@ -0,0 +1,323 @@
+      <td>Same as above but for organization</td>
+    </tr>
+    <tr>
+      <td>Update subnets from facts</td>
+      <td>Puppet</td>
+      <td>If an IP of existing interface is changed (or newly created), Foreman will try to match existing Subnet object based on this setting. The new Subnet is not created if it does not exist in Foreman, because we don't have enough information in facts, such as what the default gateway is or what is the DHCP range there</td>
+    </tr>
+    <tr>
+      <td>Exclude pattern for facts stored in foreman</td>
+      <td>Provisioning</td>
+      <td>Facts can contain a lot of data that may not be interesting. E.g. if you don't want to import all blockdevices from each host, all network interfaces etc, you can specify a pattern for ignoring. The default value contains commonly ignored patterns, e.g. "docker&ast;"</td>
+    </tr>
+    <tr>
+      <td>Ignore facts for domain</td>
+      <td>Provisioning</td>
+      <td>By default, Foreman will be creating and assigning Domain objects based on the host domain name. If you don't want this to be updated, you can disable this behavior using this setting.</td>
+    </tr>
+    <tr>
+      <td>Ignore facts for operating system</td>
+      <td>Provisioning</td>
+      <td>By default, Foreman will be creating and Operating System objects if they do not exist in Foreman yet. Also the host will be associated to such OS (either found or newly created), which is useful when OS is upgraded on the host without Foreman knowing about it. In case you e.g. upgrade RHEL 7.2 to 7.3 and there's no RHEL 7.3 on your Foreman yet, it will be created. Note that newly create OS is not fully configured for provisioning. E.g. there are no Provisioning Templates assigned to it, since we don't know which should be used. Therefore later, when you decide to reprovision the host, you may be surprised you need to provide additional configuration for the Operating System first</td>
+    </tr>
+    <tr>
+      <td>Ignore interfaces with matching identifier</td>
+      <td>Provisioning</td>
+      <td>This is similar to excluded pattern settings above, it's only limited to parsing network interfaces facts. Parsing NICs is more performance heavy operation, therefore has its own configuration. But in general, it can be superceeded by the mentioned setting.</td>
+    </tr>
+    <tr>
+      <td>Ignore Puppet facts for provisioning</td>
+      <td>Provisioning</td>
+      <td>Completely ignore networking information importing from facts, this is useful when the detection does not correctly work in your environment, e.g. you use custom names for bonds or vlan interfaces.</td>
+    </tr>
+  </tbody>
+</table>
+
+### Configuring the Ansible callback
+
+If you're an Ansible user, there is an easy way to keep your Foreman inventory up to date thanks to the foreman.py callback.
+It is also an elegant way to populate (import) unmanaged hosts to your Foreman instance. If you use the installer and install
```

Comment:
```suggestion
This is an elegant way to populate (import) unmanaged hosts to your Foreman instance. If you use the installer and install
```

Reply:
Can you import as managed hosts? And this is just a learning question, can you perform REX against unmanaged hosts?

Reply:
Yes, you can import as managed and unmanaged from compute resource. However managed hosts are then usually invalid, since they miss a lot of required information. The code simply skips all validations during import (sigh).

And yes, REX can be ran against both. In fact most of hosts that Katello deals with are unamanged (registered through sub-man). But you can apply errata on them through REX. Wait for a third blog post for more information about REX on unmanaged hosts :-)